packages/amplify-codegen/src/commands/types.js

```diff
@@ -96,22 +95,30 @@ async function generateTypes(context, forceDownloadSchema, withoutInit = false,
         const introspection = path.extname(schemaPath) === '.json';
 
         try {
-          const output = await generateTypesHelper({
-            schema,
-            queries,
-            target,
-            introspection,
-          });
-          const outputs = Object.entries(output);
-
-          const outputPath = path.join(projectPath, generatedFileName);
-          if (outputs.length === 1) {
-            const [[, contents]] = outputs;
-            fs.outputFileSync(path.resolve(outputPath), contents);
+          if (target === 'swift' && fs.existsSync(outputPath) && fs.statSync(outputPath).isDirectory()) {
+            generate(queryFilePaths, schemaPath, outputPath, '', target, '', {
+              addTypename: true,
+              complexObjectSupport: 'auto',
+            });
```

Comment:
doesn't the new `generateTypesHelper` handle multiple swift file generation yet?

Reply:
It is bugged. The `generateTypesHelper` does not set the correct filenames for the generated types. It will require a more complex fix. This solution will fix current production.